test/acs-engine-test/report/report.go

```diff
@@ -28,52 +33,69 @@ type ReportMgr struct {
+		{name: "DiskProcessingError", class: errClassDeployment, regex: "DiskProcessingError"},
+		{name: "DiskServiceInternalError", class: errClassDeployment, regex: "DiskServiceInternalError"},
+		{name: "AllocationFailed", class: errClassDeployment, regex: "AllocationFailed"},
+		{name: "NetworkingInternalOperationError", class: errClassDeployment, regex: "NetworkingInternalOperationError"},
+		{name: "PlatformFaultDomainCount", class: errClassDeployment, regex: "platformFaultDomainCount"},
+
+		{name: "K8S nodes not ready", class: errClassValidation, regex: "K8S: gave up waiting for apiserver"},
+		{name: "K8S unexpected version", class: errClassValidation, regex: "K8S: unexpected kubernetes version"},
+		{name: "K8S containers not created", class: errClassValidation, regex: "K8S: gave up waiting for containers"},
+		{name: "K8S pods not running", class: errClassValidation, regex: "K8S: gave up waiting for running pods"},
+		{name: "K8S kube-dns not running", class: errClassValidation, regex: "K8S: gave up waiting for kube-dns"},
+		{name: "K8S dashboard not running", class: errClassValidation, regex: "K8S: gave up waiting for kubernetes-dashboard"},
+		{name: "K8S kube-proxy not running", class: errClassValidation, regex: "K8S: gave up waiting for kube-proxy"},
+		{name: "K8S proxy not working", class: errClassValidation, regex: "K8S: gave up verifying proxy"},
+		{name: "K8S deployment not ready", class: errClassValidation, regex: "K8S: gave up waiting for deployment"},
+		{name: "K8S no external IP", class: errClassValidation, regex: "K8S: gave up waiting for loadbalancer to get an ingress ip"},
+		{name: "K8S nginx unreachable", class: errClassValidation, regex: "K8S: failed to get expected response from nginx through the loadbalancer"},
+
+		{name: "DCOS nodes not ready", class: errClassValidation, regex: "gave up waiting for DCOS nodes"},
+		{name: "DCOS marathon validation failed", class: errClassValidation, regex: "dcos/test.sh] marathon validation failed"},
+		{name: "DCOS marathon not added", class: errClassValidation, regex: "dcos/test.sh] gave up waiting for marathon to be added"},
+		{name: "DCOS marathon-lb not installed", class: errClassValidation, regex: "Failed to install marathon-lb"},
+
+		{name: "DockerCE failed to create network", class: errClassValidation, regex: "DockerCE: gave up waiting for network to be created"},
+		{name: "DockerCE failed to create service", class: errClassValidation, regex: "DockerCE: gave up waiting for service to be created"},
+		{name: "DockerCE service unreachable", class: errClassValidation, regex: "DockerCE: gave up waiting for service to be externally reachable"},
 	}
 }
 
-func New(jobName string, buildNum int, nDeploys int) *ReportMgr {
+func New(metricsNS, metricName, jobName string, buildNum int, nDeploys int) *ReportMgr {
 	h := &ReportMgr{}
+	h.metricsNS = metricsNS
+	h.metricName = metricName
```

Comment:
@dmitsh what would be an example of metricName that will be used in ACS Engine? E,g, in the service we have ApiExpectedFailures, ApiDuration, etc.?